webserver/cWebem.cpp

```diff
@@ -1886,16 +1804,16 @@ namespace http {
 
 		bool cWebemRequestHandler::CheckAuthentication(WebEmSession & session, const request& req, reply& rep)
 		{
-			bool bTrustedNetwork = false;
-
 			session.rights = -1; // no rights
 			session.id = "";
 			session.username = "";
 			session.auth_token = "";
+			session.istrustednetwork = false;
 
 			if (myWebem->m_userpasswords.empty())
 			{
 				_log.Log(LOG_ERROR, "No (active) users in the system! There should be at least 1 active Admin user!");
+				return false; // No users in the system!
```

Comment:
I am just thinking aloud here...
At some parts in the function we return false, because there is an error, not because of the authentication check.
Maybe before the 'return false', add a line like

`	session.reply_status = reply::internal_server_error;`

Then at the point after where '**CheckAuthentication**' is called, we check if the reply_status is anything other than 'reply::ok', and if that is the case, exit, so the user received this error

Reply:
done